Improve error message on missing root links

This is a possible error that we can obtain in logs when building and testing komodo:

```
AssertionError: The roots in the link-tree is not matching the roots defined in link_roots in dict
Roots defined: {'stable', '2019.12', 'deprecated-py2', '2021.02-py36', '2021.11-py36', 'testing', '2021.08-py36', '2022.04-py36', '2021.04-py36', '2021.06-py36', '2022.03-py36', '2022.10-py38', '2022.06-py38', 'deprecated-py36', '2021.10-py36', '2021.01-py36', '2021.12-py36', '2022.08-py38', 'deprecated', 'unstable', 'bleeding', '2021.09-py36', '2022.05-py36', '2022.01-py36', '2021.07-py36', '2020.08-py27', '2022.02-py36', '2022.09-py38', '2021.05-py36', 'stable-py36', 'stable-py2', '2022.05-py38', '2021.03-py36'}
Roots expected: {'stable', '2019.12', 'deprecated-py2', '2021.02-py36', '2021.11-py36', 'testing', '2021.08-py36', '2022.12-py38', '2022.04-py36', '2021.04-py36', '2021.06-py36', '2022.03-py36', '2022.10-py38', '2022.06-py38', 'deprecated-py36', '2022.05-py38', '2021.10-py36', '2021.01-py36', '2021.12-py36', '2022.08-py38', 'deprecated', 'unstable', 'bleeding', '2021.09-py36', '2022.05-py36', '2022.01-py36', '2021.07-py36', '2020.08-py27', '2022.02-py36', '2021.05-py36', 'stable-py36', 'stable-py2', '2022.09-py38', '2021.03-py36'}
Build step 'Execute shell' marked build as failure
```

A computer is better than a human to read the keys of these two dicts and say more precisely what the mismatch is.

--- a/komodo/symlink/sanity_check.py
+++ b/komodo/symlink/sanity_check.py
@@ -82,10 +82,11 @@ def assert_root_nodes(link_dict):
     inferred_roots = _get_root_nodes(link_dict)
     if set(input_roots) != inferred_roots:
         raise AssertionError(
-            "The roots in the link-tree is not matching "
-            + "the roots defined in link_roots in dict\n"
+            "The roots in the link-tree do not match "
+            + "the roots defined in the root_links dict\n"
             + f"Roots defined: {set(input_roots)}\n"
-            + f"Roots expected: {inferred_roots}",
+            + f"Roots expected: {inferred_roots}\n"
+            + f"Missing root(s): {str(set(inferred_roots).difference(input_roots))}"
         )
 
 

--- a/tests/test_sanity_check.py
+++ b/tests/test_sanity_check.py
@@ -1,0 +1,22 @@
+import pytest
+
+from komodo.symlink.sanity_check import assert_root_nodes
+
+
+def test_assert_root_nodes_error_message():
+    link_dict = {
+        "links": {
+            "stable": "2012.01",
+            "testing": "2012.03",
+            "missing_root_1": "2011.12",
+            "missing_root_2": "2011.11",
+            "missing_root_3": "2011.10",
+        },
+        "root_links": ["stable", "testing"],
+    }
+
+    with pytest.raises(
+        AssertionError,
+        match=r"Missing root\(s\): {(?=.*missing_root_1)(?=.*missing_root_2)(?=.*missing_root_3)",
+    ):
+        assert_root_nodes(link_dict)

